### PR TITLE
[prototype][glossary] Example of including glossary fragment in concepts page

### PR DESCRIPTION
**Pull request by abiogenesis-now**
@chenopis @Bradamant3 PTAL
cc @heidecke 

Trying to see if we can get this into the glossary branch so we can show to SIG Docs how it ought to work.

The actual content ("statefulset") isn't fixed/revised in any way, I mainly wanted to get the mechanics of the snippets working, more so than that aspect (for now).

The way I approached this is a little different (Jekyll has some limitations, so I scoured their docs for ways to try to make this as efficient as possible during build-time and as pleasant to read/script).

Basically this PR:
* Sets up a `glossary` directory within `_data`, where each term gets its own YAML file. The most controversial part of this is including multi-line markdown as one of the YAML fields, but with the `markdownify` filter it seems to work fine.
* The reason I did this is that it makes it really easy to index into a specific term (`site.data.glossary.statefulset`) as well as to get all items (`site.data.glossary`). (Iteration over an arbitrary subdirectory, e.g. something in `/_includes`, doesn't appear to be supported).
* The template itself is `_includes/templates/glossary.md`. It gets applied to a specific glossary term based on `include.term`.
* I also foresee a need to be able to easily access the tag-to-terms mapping (as opposed to term-to-tags). `tag-map.md` calculates this out and stores the results in the form: `[[<term>, <tags>], [<term2>, <tags2>], ...]`. You can kind of treat this like a global variable. By including `tag-map.md` with the `include_cached` command, we can avoid recomputing it multiple times during the site build, so it's more efficient.

TLDR the stuff above is some things that I thought through, and I've shown that it works on the StatefulSet page.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4949)

### CODE REVIEW

**Comment by abiogenesis-now** [comment]
btw @chenopis I ended up adding the Go file we discussed that checks that all glossary terms have tags from the canonical set. 

I've confirmed that it works (running tests locally). Let me know what you think of this PR's overall approach! it should apply to the rest of the glossary terms and then we can start worrying about the kubernetes-specific conceptual stuff

**Comment by chenopis** [comment]
@abiogenesis-now Oops! I needed to setup the preview deploy. You can view the preview here:

https://kubernetes-io-glossary-project.netlify.com/

**Comment by chenopis** [comment]
@abiogenesis-now Just fix the merge conflict, and I think we're good to go!

File: _data/glossary/_example.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+- Tangential Term
+- Commonly Used With
+tags:
+- Some Tag
+short-description: |
+  Required. One or two lines that provide a minimum definition. Do not repeat the term. Prefer fragments. Model after tooltips.
+long-description: |
+  Optional. Longer additional text to appear after (in conjunction with) short description. Provide where the short description is not sufficient as the intro paragraph to a topic. Write complete but concise sentences.
+
+  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed hendrerit dictum nisi quis posuere. Integer eleifend neque lobortis ultrices viverra. Praesent viverra placerat ex sit amet aliquet. Aliquam non ipsum ut lectus viverra pulvinar id vitae nisi. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Suspendisse aliquet congue tristique. Pellentesque at condimentum orci, iaculis pellentesque urna. Aliquam ut rhoncus lorem, a tristique est. Quisque eu est sem.
```
Comment (chenopis) [inline _data/glossary/_example.yml]: You can kill the lorem ipsum placeholder text.

File: _data/glossary/statefulset.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+related:
+- Deployment
+- Pod
+tags:
+- Storage
+- Workload
+- API Object
+short-description: |
+  A Controller that provides a unique identity to its Pods. It provides guarantees about the ordering of deployment and scaling.
+long-description: |
```
Comment (chenopis) [inline _data/glossary/statefulset.yml]: BTW, can we see an example of the `long-description` being used somewhere? Thx

**Comment by abiogenesis-now** [comment]
oops @chenopis I accidentally pushed my branch to your branch on upstream instead of to this branch >< will avoid that in the future, but this is PR is no longer necessary so I'm closing it

**Comment by chenopis** [comment]
@abiogenesis-now No worries. It's basically the same as merging it here. :)
